tests/gatewayapitests/npltests/gateway_npl_test.go

```diff
@@ -0,0 +1,453 @@
+)
+
+var ctrl *akogatewayapik8s.GatewayController
+
+const (
+	DEFAULT_NAMESPACE = "default"
+)
+
+func TestMain(m *testing.M) {
+
+	tests.KubeClient = k8sfake.NewSimpleClientset()
+	tests.GatewayClient = gatewayfake.NewSimpleClientset()
+	integrationtest.KubeClient = tests.KubeClient
+
+	// Sets the environment variables
+	os.Setenv("CLUSTER_NAME", "cluster")
+	os.Setenv("CLOUD_NAME", "CLOUD_VCENTER")
+	os.Setenv("SEG_NAME", "Default-Group")
+	os.Setenv("POD_NAMESPACE", utils.AKO_DEFAULT_NS)
+	os.Setenv("FULL_SYNC_INTERVAL", utils.AKO_DEFAULT_NS)
+	os.Setenv("ENABLE_EVH", "true")
+	os.Setenv("TENANT", "admin")
+	os.Setenv("POD_NAME", "ako-0")
+	os.Setenv("SERVICE_TYPE", "NodePortLocal")
+
+	// Set the user with prefix
+	_ = lib.AKOControlConfig()
+	lib.SetAKOUser(akogatewayapilib.Prefix)
+	lib.SetNamePrefix(akogatewayapilib.Prefix)
+	lib.AKOControlConfig().SetIsLeaderFlag(true)
+	akoControlConfig := akogatewayapilib.AKOControlConfig()
+	akoControlConfig.SetEventRecorder(lib.AKOGatewayEventComponent, tests.KubeClient, true)
+	registeredInformers := []string{
+		utils.ServiceInformer,
+		utils.SecretInformer,
+		utils.NSInformer,
+		utils.PodInformer,
+	}
+	utils.AviLog.SetLevel("DEBUG")
+	utils.NewInformers(utils.KubeClientIntf{ClientSet: tests.KubeClient}, registeredInformers, make(map[string]interface{}))
+	data := map[string][]byte{
+		"username": []byte("admin"),
+		"password": []byte("admin"),
+	}
+	object := metav1.ObjectMeta{Name: "avi-secret", Namespace: utils.GetAKONamespace()}
+	secret := &corev1.Secret{Data: data, ObjectMeta: object}
+	tests.KubeClient.CoreV1().Secrets(utils.GetAKONamespace()).Create(context.TODO(), secret, metav1.CreateOptions{})
+
+	akoApi := integrationtest.InitializeFakeAKOAPIServer()
+	defer akoApi.ShutDown()
+
+	tests.NewAviFakeClientInstance(tests.KubeClient)
+	defer integrationtest.AviFakeClientInstance.Close()
+
+	ctrl = akogatewayapik8s.SharedGatewayController()
+	ctrl.DisableSync = false
+	ctrl.InitGatewayAPIInformers(tests.GatewayClient)
+	akoControlConfig.SetGatewayAPIClientset(tests.GatewayClient)
+
+	stopCh := utils.SetupSignalHandler()
+	ctrlCh := make(chan struct{})
+	quickSyncCh := make(chan struct{})
+
+	waitGroupMap := make(map[string]*sync.WaitGroup)
+	wgIngestion := &sync.WaitGroup{}
+	waitGroupMap["ingestion"] = wgIngestion
+	wgFastRetry := &sync.WaitGroup{}
+	waitGroupMap["fastretry"] = wgFastRetry
+	wgSlowRetry := &sync.WaitGroup{}
+	waitGroupMap["slowretry"] = wgSlowRetry
+	wgGraph := &sync.WaitGroup{}
+	waitGroupMap["graph"] = wgGraph
+	wgStatus := &sync.WaitGroup{}
+	waitGroupMap["status"] = wgStatus
+
+	integrationtest.AddConfigMap(tests.KubeClient)
+	go ctrl.InitController(k8s.K8sinformers{Cs: tests.KubeClient}, registeredInformers, ctrlCh, stopCh, quickSyncCh, waitGroupMap)
+	os.Exit(m.Run())
+
+}
+
+const (
+	defaultPodName  = "test-pod"
+	defaultNodeName = "test-node"
+	defaultPodPort  = 80
+	defaultHostIP   = "10.10.10.10"
+	defaultPodIP    = "192.168.32.10"
+	defaultNodePort = 40001
```

Comment:
This might also need to be changed to 6000x

Reply:
changed to 60001